### ISSUE TITLE
Application not detected as started when `spring-aot-maven-plugin` is active

### ISSUE THREAD

**Issue by ch4mpy** [spring-projects-experimental/spring-native#740]
My apps are not detected as started anymore when I add `spring-aot-maven-plugin` to my pom file.

This has at least to manifestations :
- spinner never changes to green arrow in Spring Boot Dashboard
- more annoying : `springdoc-openapi-maven-plugin` fails my builds as it relies on `spring-boot-maven-plugin` to start the app at `pre-integration-test` and waits for the app to be reported as started before fetching OpenAPI spec file.

**Comment by sdeleuze** [comment]
Could you please provide a project that would allow us to reproduce this issue?

**Comment by ch4mpy** [comment]
Here is a repo with minimal project : git@github.com:ch4mpy/spring-native-740.git

Steps to reproduce: 

1. vanilla project unzipped from start.spring.io reproduces the issue I'm reporting : just click "start" in STS Boot Dashboard and the spinner never turns into green arrow
2. comment spring-aot-maven-plugin in [pom file](https://github.com/ch4mpy/spring-native-740/blob/6d4627bbe4cbe7cf2d7f278060792bf2eb271479/pom.xml#L88) and the problem goes
3. add [springdoc-openapi-maven-plugin](https://github.com/ch4mpy/spring-native-740/blob/6d4627bbe4cbe7cf2d7f278060792bf2eb271479/pom.xml#L107), spring-aot-maven-plugin still being commented out => build OK
4. re-activate spring-aot-maven-plugin => build fails because app never detected as started

**Comment by ch4mpy** [comment]
I just noticed that `spring-aot-maven-plugin` isn't necessary to fail maven build. Just add :
``` xml
				<executions>
					<execution>
						<id>pre-integration-test</id>
						<goals>
							<goal>start</goal>
						</goals>
					</execution>
					<execution>
						<id>post-integration-test</id>
						<goals>
							<goal>stop</goal>
						</goals>
					</execution>
				</executions>
```

to what you get from `start.spring.io` with spring-native and web dependencies, and build fails

**Comment by sdeleuze** [comment]
Maybe related to the fact we disable MBeans for footprint reasons, since I see a `Failed to connect to MBean server at port 9001` message when running `mvn pre-integration-test`.

**Comment by ch4mpy** [comment]
How can I enable it ?

Setting following properties in application.properties does not improve the situation :
```
spring.jmx.enabled=true
server.tomcat.mbeanregistry.enabled=true
```

**Comment by sdeleuze** [comment]
@ch4mpy Could you set `removeJmxSupport = false` on Spring AOT plugin side and let me know how it goes? Maybe we could infer its configuration from `spring.jmx.enabled` one, not sure yet.

**Comment by spring-projects-issues** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by deodorel** [comment]
the issue was in fact caused by having the spring-aot dependency in one of the maven modules (not the one with the application class). I am closing it. Thanks for looking into it.

**Comment by sdeleuze** [comment]
Should be solved by cc87cf548e06cf4534af13dee612a2e9768bded8.
